BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c

```diff
@@ -420,6 +424,13 @@ UpdateFwst (
     FwstAcpiTablePtr->EsrtTableEntry[Count].FwType = ESRT_FW_TYPE_SYSTEMFIRMWARE;
     FwstAcpiTablePtr->EsrtTableEntry[Count].LastAttemptVersion = FwUpdCompStatus->LastAttemptVersion;
     FwstAcpiTablePtr->EsrtTableEntry[Count].LastAttemptStatus = FwUpdCompStatus->LastAttemptStatus;
+    if (CompareGuid(&(FwUpdCompStatus->FirmwareId), &gSblFWUpdateImageFileGuid) == TRUE) {
+      Version = ((VerInfoTbl->ImageVersion.CoreMajorVersion << 24) | (VerInfoTbl->ImageVersion.CoreMinorVersion << 16) | \
+                (VerInfoTbl->ImageVersion.ProjMajorVersion << 8) | (VerInfoTbl->ImageVersion.ProjMinorVersion));
+      FwstAcpiTablePtr->EsrtTableEntry[Count].FwVersion = Version;
+    } else {
+      PlatformUpdateCurrentVersion(&(FwUpdCompStatus->FirmwareId), &(FwstAcpiTablePtr->EsrtTableEntry[Count].FwVersion));
```

Comment:
All platforms must have this function even if this is not used. What about creating a VersionLib? Each platforms set version info for each fw components and UpdateFwst gets version thru VersionLib.